website/docs/guides/delivery.mdx

```diff
@@ -0,0 +1,473 @@
+```yaml title="test/canary.yaml"
+apiVersion: flagger.app/v1beta1
+kind: Canary
+metadata:
+  name: podinfo
+spec:
+  # deployment reference
+  targetRef:
+    apiVersion: apps/v1
+    kind: Deployment
+    name: podinfo
+  # HPA reference (optional)
+  autoscalerRef:
+    apiVersion: autoscaling/v2beta2
+    kind: HorizontalPodAutoscaler
+    name: podinfo
+  # the maximum time in seconds for the canary deployment
+  # to make progress before it is rollback (default 600s)
+  progressDeadlineSeconds: 60
+  service:
+    # ClusterIP port number
+    port: 9898
+    # container port number or name (optional)
+    targetPort: 9898
+  analysis:
+    # schedule interval (default 60s)
+    interval: 30s
+    # max number of failed metric checks before rollback
+    threshold: 5
+    # max traffic percentage routed to canary
+    # percentage (0-100)
+    maxWeight: 50
+    # canary increment step
+    # percentage (0-100)
+    stepWeight: 5
+    # Linkerd Prometheus checks
+    metrics:
+    - name: request-success-rate
+      # minimum req success rate (non 5xx responses)
+      # percentage (0-100)
+      thresholdRange:
+        min: 99
+      interval: 1m
+    - name: request-duration
+      # maximum req duration P99
+      # milliseconds
+      thresholdRange:
+        max: 500
+      interval: 30s
+    # testing (optional)
+    webhooks:
+      - name: acceptance-test
+        type: pre-rollout
+        url: http://loadtester.flagger/
+        timeout: 30s
+        metadata:
+          type: bash
+          cmd: "curl -sd 'test' http://podinfo-canary.test:9898/token | grep token"
+      - name: load-test
+        type: rollout
+        url: http://loadtester.flagger/
+        metadata:
+          cmd: "hey -z 2m -q 10 -c 2 http://podinfo-canary.test:9898/"
+```
+
+Finally, add a Kustomization file to apply all resources to the `test` namespace:
+
+```yaml title="test/kustomization.yaml"
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+namespace: test
+resources:
+- namespace.yaml
+- deployment.yaml
+- canary.yaml
+```
+
+After a short time, the status of the canary object should be set to `Initialized`:
+
+```bash
+> kubectl -n test get canary podinfo
+NAME      STATUS        WEIGHT   LASTTRANSITIONTIME
+podinfo   Initialized   0        2022-07-19T11:57:17Z
+```
+
+Now trigger a new rollout by bumping the version of `podinfo`:
+
+```bash
+> kubectl -n test set image deployment/podinfo podinfod=ghcr.io/stefanprodan/podinfo:6.0.1
+```
+
+During the progressive rollout, the canary object reports on its current status:
+
+```bash
+> k -n test get canary podinfo
+NAME      STATUS        WEIGHT   LASTTRANSITIONTIME
+podinfo   Progressing   35       2022-07-19T12:25:04
+```
+
+After a short time the rollout is completed and the status of the canary object is set to `Succeeded`:
+
+```bash
+> kubectl -n test get canary podinfo
+NAME      STATUS      WEIGHT   LASTTRANSITIONTIME
+podinfo   Succeeded   0        2022-07-19T12:28:04Z
+```
```

Comment:
Add something to end it maybe:

##Summary
Congratulations, you have now completed a progressive delivery rollout with Flagger and Linkerd :tada: 

Next steps:
- Explore more of what [Flagger](https://flagger.app) can offer
- Configure manual approving for progressive deliver deployments (link to guide)